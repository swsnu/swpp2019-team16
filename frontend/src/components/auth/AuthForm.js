import React from 'react';
import styled from 'styled-components'
import Button from '../common/Button/Button';
import TextInput from '../common/TextInput';

const AuthFormBlock = styled.div``;

AuthForm.propTypes = {};

function AuthForm({ form, onChange, onSubmit }) {
  return (
    <AuthFormBlock>
        <div>
        <TextInput
            label='username'
            type="text"
            id="email-input"
            name="username"
            margin='normal'
            variant='outlined'
            onChange={onChange}
            value={form.username}
        />
        </div>        
        <div>
        <TextInput
            label='password'
            type="password"
            id="pw-input"
            name="password"
            margin='normal'
            variant='outlined'
            onChange={onChange}
            value={form.password}
        />
        </div>
        <Button id="login-button" onClick={onSubmit} children='login'/>
        <Button id="signup-button" onClick={onSubmit} children='sign up'/>
    </AuthFormBlock>
  );
}

export default AuthForm;