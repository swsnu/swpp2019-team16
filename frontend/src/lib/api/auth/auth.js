import client from '../client';

export function login({ email, password }) {
  return client.post('/api/v1/user/login', { email, password });
}

export function register({ userType, email, password, carType, plateNo }) {
  return client.post('/api/v1/user/register', {
    userType,
    email,
    password,
    carType,
    plateNo,
  });
}
