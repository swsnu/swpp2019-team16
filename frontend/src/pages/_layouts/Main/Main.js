import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import HeaderContainer from '../../../containers/common/HeaderContainer/HeaderContainer';
import theme from '../../../lib/styles/theme';
import ThemeProvider from '@material-ui/styles/ThemeProvider';
import { CssBaseline } from '@material-ui/core';

const Root = styled.div`
  display: flex;
  flex-direction: column;
`;

Main.propTypes = {
  children: PropTypes.node.isRequired,
};

function Main({ children }) {
  return (
    <ThemeProvider theme={theme}>
      {/* <CssBaseline /> enables lib/material-ui/theme to work */}
      <CssBaseline />
      <Root>
        <HeaderContainer />
        {children}
      </Root>
    </ThemeProvider>
  );
}

export default Main;
