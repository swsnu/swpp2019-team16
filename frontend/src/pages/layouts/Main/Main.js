import React from 'react';
import styled from 'styled-components';
import Header from '../../../components/common/Header';

const MainBlock = styled.div``;

Main.propTypes = {};

function Main() {
  return (
    <MainBlock>
    </MainBlock>
  );
}

export default Main;
