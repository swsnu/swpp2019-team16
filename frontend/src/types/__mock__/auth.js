export const mockRiderAuth = {
  id: 1,
  email: 'rider1@gmail.com',
  user_type: 'rider',
  vehicle: {
    car_type: '',
    plate_no: '',
  },
};

export const mockDriverAuth = {
  id: 1,
  email: 'driver1@gmail.com',
  user_type: 'driver',
  vehicle: {
    car_type: '',
    plate_no: '',
  },
};
