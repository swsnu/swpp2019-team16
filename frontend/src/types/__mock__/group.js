import { createMockRider, mockDriver } from './user';

export const mockGroup = {
  groupId: 1,
  driver: mockDriver,
  riders: [
    createMockRider(1, 1, 'a@gmail.com'),
    createMockRider(2, 1, 'b@gmail.com'),
    createMockRider(3, 1, 'c@gmail.com'),
  ],
  fromLocation: 'A',
  toLocation: 'B',
};
