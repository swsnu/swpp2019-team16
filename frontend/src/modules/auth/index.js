import auth from './auth';

export * from './auth';
export default auth;
