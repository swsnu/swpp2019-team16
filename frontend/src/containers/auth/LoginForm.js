import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { changeField, initalizeForm, login } from '../../modules/auth';
import AuthForm from '../../components/auth/AuthForm';
import {check} from '../../modules/user';
import {withRouter} from 'react-router-dom';

LoginForm.propTypes = {};

function LoginForm({ history }) {

  const dispatch = useDispatch();

  const { form, auth, authError, user } = useSelector(({ auth, user }) => ({
    form: auth.login,
    auth: auth.auth,
    authError: auth.authError,
    user: user.user,
  }));

  const onChange = e => {
    const { value, name } = e.target;
    dispatch(
      changeField({
        form: 'login',
        key: name,
        value,
      }),
    );
  };

  const onSubmit = e => {
    e.preventDefault();
    const { username, password } = form;
    dispatch(login({ username, password }));
  };

  useEffect(() => {
    dispatch(initalizeForm('login'));
  }, [dispatch]);

  useEffect(() => {
    if (authError) {
      alert('Email or password is wrong');
      return;
    }

    if (auth) {
      dispatch(check());
    }
  }, [auth, authError, dispatch]);

  useEffect(() => {
    if (user) {
      history.push('/request');
      try {
        localStorage.setItem('user', JSON.stringify(user));
      } catch (e) {
        console.log('localStorage is not working');
      }
    }
  }, [history, user]);
  return (
    <AuthForm
      form={form}
      onChange={onChange}
      onSubmit={onSubmit}
    />
  );
}

export default withRouter(LoginForm);