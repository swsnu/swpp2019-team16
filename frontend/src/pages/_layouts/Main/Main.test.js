import React from 'react';
import { renderWithRedux } from '../../../test/utils';
import { MemoryRouter, Route } from 'react-router-dom';
import Main from './Main';
import { mockRiderAuth } from '../../../types/__mock__/auth';

describe('<Main />', () => {
  const state = {
    auth: {
      auth: mockRiderAuth,
    },
  };
  const children = { children: 1 };
  it('SHOULD match with snapshot', async () => {
    const { container } = renderWithRedux(
      <MemoryRouter initialEntries={['/']}>
        <Route component={Main} path="/" />
      </MemoryRouter>,
      state,
    );
    expect(container).toMatchSnapshot();
  });
});
