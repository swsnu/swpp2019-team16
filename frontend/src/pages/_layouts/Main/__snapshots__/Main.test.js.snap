// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> SHOULD match with snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa bPGLpq"
  >
    <div
      class="makeStyles-root-1"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <h4
          class="MuiTypography-root makeStyles-title-3 MuiTypography-h4"
        >
          Ya-Ta!
        </h4>
        <button
          class="MuiButtonBase-root MuiButton-root makeStyles-root-38 makeStyles-root-41 MuiButton-text makeStyles-text-40 makeStyles-text-43 MuiButton-text makeStyles-text-40 makeStyles-text-43"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Log In
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
